include/deal.II/base/utilities.h

```diff
@@ -1498,6 +1523,11 @@ namespace Utilities
   T
   unpack(const std::vector<char> &buffer, const bool allow_compression)
   {
+    // see if the object is empty. if so, shortcut the whole
+    // operation and return an empty object.
+    if (buffer.size() == 0)
+      return T();
```

Comment:
This seems wrong - we should throw an exception here like we do below (e.g., what if `T` is not default-constructible?)

Reply:
same here